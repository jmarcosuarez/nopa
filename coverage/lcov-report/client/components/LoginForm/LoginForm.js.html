<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/LoginForm/LoginForm.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">client/components/LoginForm</a> LoginForm.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/5</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { TextInput, Button } from '../';
import { isValid } from '../../containers/Login/Revalidation';
import styles from './LoginForm.css';
&nbsp;
const LoginForm = <span class="fstat-no" title="function not covered" >({</span>
  form,
  onChange,
  updateAndValidate,
  validate,
  valid,
  errors = <span class="branch-0 cbranch-no" title="branch not covered" >{},</span>
  onSubmit,
  submitCb,
}) =&gt;
  (
<span class="cstat-no" title="statement not covered" >    &lt;div className={styles.form}&gt;</span>
      &lt;TextInput 
        name="surname"
        type="text"
        className={isValid(errors.surname) ? '' : 'error'}
        value={form.surname}
        onChange={updateAndValidate}
        errors={errors.surname}
      /&gt;
      &lt;TextInput 
        name="sortCode"
        type="text"
        className={isValid(errors.sortCode) ? '' : 'error'}
        value={form.sortCode}
        onChange={onChange}
        errors={errors.sortCode}
      /&gt;
      &lt;TextInput 
        name="accountNumber"
        type="text"
        className={isValid(errors.accountNumber) ? '' : 'error'}
        value={form.accountNumber}
        onChange={onChange}
        errors={errors.accountNumber}
      /&gt;
      &lt;TextInput 
        name="passCode"
        type="text"
        className={isValid(errors.passCode) ? '' : 'error'}
        value={form.passCode}
        onChange={onChange}
        errors={errors.passCode}
      /&gt;
      &lt;TextInput 
        name="memorableWord"
        type="text"
        className={isValid(errors.memorableWord) ? '' : 'error'}
        value={form.memorableWord}
        onChange={updateAndValidate}
        errors={errors.memorableWord}
      /&gt;
      &lt;Button onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >onSubmit(submitCb)}</span>&gt;LogIn &amp; Connect&lt;/Button&gt;
    &lt;/div&gt;
  );
&nbsp;
LoginForm.propTypes = {
  form: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  updateAndValidate: PropTypes.func.isRequired,
  validate: PropTypes.func.isRequired,
  valid: PropTypes.func,
  onSubmit: PropTypes.func.isRequired,
  submitCb: PropTypes.func.isRequired,
  errors: PropTypes.object,
};
&nbsp;
LoginForm.defaultProps = {
  errors: {},
  valid: <span class="fstat-no" title="function not covered" >()</span> =&gt; {},
};
&nbsp;
export default LoginForm;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 16:15:11 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
