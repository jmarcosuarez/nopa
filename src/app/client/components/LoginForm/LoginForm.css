@import "../../../static/stylesheets/abstracts/all.css";

.form {
  padding: 1rem 1rem 2rem 1rem;
  margin-top: 2rem;
  margin-right: auto;
  margin-left: auto;
  /*background: #fff;*/
  max-width: 25rem;
  box-shadow: 0 15px 35px rgba(50,50,93,.1),0 5px 15px rgba(0,0,0,.07);
}

.btn {
 display: inline-block;
 padding: .75rem 2rem;
 margin-top: 1.618rem;
 font-weight: 400;
 text-align: center;
 text-transform: uppercase;
 color: #fff;
 vertical-align: middle;
 white-space: nowrap;
 background-color: var(--color-lightblue);
 border: 1px solid transparent;
 box-shadow: 0 15px 35px rgba(50,50,93,.1),0 5px 15px rgba(0,0,0,.07);
 cursor: pointer;
 user-select: none;
 /*@include transition;*/
 
 &:focus,
 &:hover {
  background-color: color(var(--color-lightblue) a(13%)); /* As rgba(255, 0, 0, .13)*/
  box-shadow: 0 18px 35px rgba(50,50,93,.1),0 8px 15px rgba(0,0,0,.07);
 }
 
 &:focus {
  outline: 0;
 }
}

