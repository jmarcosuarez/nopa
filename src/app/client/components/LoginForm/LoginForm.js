import React from 'react';
import PropTypes from 'prop-types';
import { TextInput } from '../';
import { isValid } from '../../containers/Login/Revalidation';
import styles from './LoginForm.css';

const LoginForm = ({
  form,
  onChange,
  updateAndValidate,
  validate,
  valid,
  errors = {},
  onSubmit,
  submitCb,
}) =>
  (
    <div className={styles.form}>
      <TextInput 
        name="surname"
        type="text"
        className={isValid(errors.surname) ? '' : 'error'}
        value={form.surname}
        onChange={updateAndValidate}
        errors={errors.surname}
      />
      <TextInput 
        name="sortCode"
        type="text"
        className={isValid(errors.sortCode) ? '' : 'error'}
        value={form.sortCode}
        onChange={onChange}
        errors={errors.sortCode}
      />
      <TextInput 
        name="accountNumber"
        type="text"
        className={isValid(errors.accountNumber) ? '' : 'error'}
        value={form.accountNumber}
        onChange={onChange}
        errors={errors.accountNumber}
      />
      <TextInput 
        name="passCode"
        type="text"
        className={isValid(errors.passCode) ? '' : 'error'}
        value={form.passCode}
        onChange={onChange}
        errors={errors.passCode}
      />
      <TextInput 
        name="memorableWord"
        type="text"
        className={isValid(errors.memorableWord) ? '' : 'error'}
        value={form.memorableWord}
        onChange={updateAndValidate}
        errors={errors.memorableWord}
      />
      <button className={styles.btn} onClick={() => onSubmit(submitCb)}>Submit
    </button>
    </div>
  );

LoginForm.propTypes = {
  form: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  updateAndValidate: PropTypes.func.isRequired,
  validate: PropTypes.func.isRequired,
  valid: PropTypes.func,
  onSubmit: PropTypes.func.isRequired,
  submitCb: PropTypes.func.isRequired,
  errors: PropTypes.object,
};

LoginForm.defaultProps = {
  errors: {},
  valid: () => {},
};

export default LoginForm;
