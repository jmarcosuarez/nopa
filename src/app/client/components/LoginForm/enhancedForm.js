import React from 'react';

const enhancedForm = Component => 
  class HigherOrderComponent extends React.Component {

    constructor(props) {
      super(props);
      // this.state = { form: props.initialState };
      this.state = {
        form: {
          surname: '',
          sortCode: '', 
          accountNumber: '',
          passCode: '',
          memorableWord: '',
        },
      };
      this.handleSubmit = this.handleSubmit.bind(this);
      this.handleChange = this.handleChange.bind(this);
    }
    handleSubmit(e) {
      e.preventDefault();
      console.log(JSON.stringify(this.state, null, 4));
    }

    handleChange(e) {
      const target = e.target;
      const name = target.name;
      const value = target.type === 'checkbox' 
      ? target.checked 
      : target.value;
      
      this.setState(state => ({ form: { ...state.form, ...{ [name]: value } } }));    
    }
    render() {
      const { initialState, ...rest } = this.props;
      const { form } = this.state;
      return React.createElement(Component, {
        state: form,
        handleSubmit: this.handleSubmit,
        handleChange: this.handleChange,
      });
    }
};

export default enhancedForm;
