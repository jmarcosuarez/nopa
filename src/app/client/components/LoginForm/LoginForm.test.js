/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';

test('should render correctly', () => {
  const fc = jest.fn();
  const object = { 
    bank: {
      choosenBank: "HCBC",
      transactions: [
        { id: 1, dateStr: "Now", beneficary: "PAYPAL ZARA", value: "- Â£35.98" },
      ],
    },
  };
  const wrapper = shallow(
    <LoginForm
      form={object}
      value="memorableWorld"
      onChange={fc}
      updateAndValidate={fc}
      validate={fc}
      valid={fc}
      onSubmit={fc}
      submitCb={fc}
      errors={object}
    />,
);
  expect(toJson(wrapper)).toMatchSnapshot();
});
