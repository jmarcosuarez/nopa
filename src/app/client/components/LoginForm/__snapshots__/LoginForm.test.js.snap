// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="form"
>
  <TextInput
    className=""
    errors={Array []}
    name="surname"
    onChange={[Function]}
    type="text"
  />
  <TextInput
    className=""
    errors={Array []}
    name="sortCode"
    onChange={[Function]}
    type="text"
  />
  <TextInput
    className=""
    errors={Array []}
    name="accountNumber"
    onChange={[Function]}
    type="text"
  />
  <TextInput
    className=""
    errors={Array []}
    name="passCode"
    onChange={[Function]}
    type="text"
  />
  <TextInput
    className=""
    errors={Array []}
    name="memorableWord"
    onChange={[Function]}
    type="text"
  />
  <Button
    onClick={[Function]}
  >
    LogIn & Connect
  </Button>
</div>
`;
